test/test.js

```diff
@@ -107,13 +108,33 @@ describe('delegated event listeners', function() {
         order.push(2);
       };
 
-      on('test:order', '.js-test-parent', one);
-      on('test:order', '.js-test-child', two);
+      const three = function(event) {
+        assert.strictEqual(child, event.target);
+        assert.strictEqual(parent, event.currentTarget);
+        assert.strictEqual(this, event.currentTarget);
+        assert.strictEqual(this, parent);
+        order.push(3);
+      };
+
+      const four = function(event) {
+        assert.strictEqual(child, event.target);
+        assert.strictEqual(child, event.currentTarget);
+        assert.strictEqual(this, event.currentTarget);
+        assert.strictEqual(this, child);
+        order.push(4);
+      };
+
+      on('test:order', '.js-test-parent', one, {capture: true});
+      on('test:order', '.js-test-child', two, {capture: true});
+      on('test:order', '.js-test-parent', three);
+      on('test:order', '.js-test-child', four);
       fire(this.child, 'test:order');
-      off('test:order', '.js-test-parent', one);
-      off('test:order', '.js-test-child', two);
+      off('test:order', '.js-test-parent', one, {capture: true});
+      off('test:order', '.js-test-child', two, {capture: true});
+      off('test:order', '.js-test-parent', three);
+      off('test:order', '.js-test-child', four);
 
-      assert.deepEqual([2, 1], order);
+      assert.deepEqual([1, 2, 4, 3], order);
```

Comment:
Cool, I think this is right now. Start at **1** parent, walk down to **2** child, visit **4** child again, then walk up to **3** parent.